scripts/pi-hole/js/messages.js

```diff
@@ -211,40 +262,63 @@ $(function () {
   });
 });
 
+// Remove 'bnt-group' class from container, to avoid grouping
+$.fn.dataTable.Buttons.defaults.dom.container.className = "dt-buttons";
+
 function deleteMessage() {
-  var tr = $(this).closest("tr");
-  var id = tr.attr("data-id");
+  // Passes the button data-del-id attribute as ID
+  var ids = [parseInt($(this).attr("data-del-id"), 10)];
+  delMsg(ids);
+}
+
+function delMsg(ids) {
+  // Check input validity
+  if (!Array.isArray(ids)) return;
+
+  // Finding out which <tr> should be deleted, based on ID
+  var trs = [];
+  for (var id in ids) {
+    if (Object.hasOwnProperty.call(ids, id)) {
+      // Exploit prevention: Return early for non-numeric IDs
+      if (typeof ids[id] !== "number") return;
+      trs.push(ids[id]);
+    }
+  }
 
   utils.disableAll();
-  utils.showAlert("info", "", "Deleting message with ID " + parseInt(id, 10), "...");
+  var idstring = JSON.stringify(ids);
```

Comment:
I notice you are creating a filtered array (`trs`) with only numeric values from `ids`, but on line 289 you are using the unfiltered variable. I think you should use:

```js
var idstring = JSON.stringify(trs);
```

Reply:
It doesn't matter at all. If there is a single non-numeric entry in `ids`, the entire function will return early and never reach this point.